### PR TITLE
Add cache busting

### PR DESCRIPTION
**Pull request by asaunier**
GMF 2 also uses Pyramid 1.6 and its cache busting system.

I have then ported the GMF code
https://github.com/camptocamp/c2cgeoportal/blob/master/c2cgeoportal/__init__.py#L344-L352
https://github.com/camptocamp/c2cgeoportal/blob/master/c2cgeoportal/lib/cacheversion.py
to our project with a few adaptations (python3, no server caching yet).

It works pretty nice and almost out of the box.

Instead of a random uuid string, I would like to use the short last commit hash provided by https://github.com/c2corg/v6_ui/blob/cache_buster2/config/default#L4
but I don't know how to get a config parameter from within a standard function (say ``get_cache_version``). I have tried to use some ``CACHE_VERSION`` variable set in the ``main()`` function but it's then not available in ``get_cache_version``. flake8 by the way says that:
> c2corg_ui/__init__.py:3:1: F401 'CACHE_VERSION' imported but unused
c2corg_ui/__init__.py:14:5: F811 redefinition of unused 'CACHE_VERSION' from line 3
c2corg_ui/__init__.py:14:5: F841 local variable 'CACHE_VERSION' is assigned to but never used

Anyone knows what's the best way to do this? Thanks!

Fixes https://github.com/c2corg/v6_ui/issues/82

### CODE REVIEW

File: c2corg_ui/lib/cacheversion.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# TODO cache
+def get_cache_version():
+    """Return a cache version that is regenerate after each cache invalidation"""
+    # return CACHE_VERSION
+    return uuid.uuid4().hex
+
+
+def version_cache_buster(request, subpath, kw):  # pragma: nocover
+    return urljoin(get_cache_version() + "/", subpath), kw
```
Comment (tsauerwein) [inline c2corg_ui/lib/cacheversion.py]: Here you can get the Git revision with:

    request.registry.settings.get('cache_version')

**Comment by tsauerwein** [comment]
This looks good. How does it work in development mode with `make serve`? The cache is not used in that case?

**Comment by gberaudo** [comment]
We need to do something to prevent cache poisoning, see https://github.com/camptocamp/schweizmobil_re3/issues/516. I can have a look at it before merging.

**Comment by tsauerwein** [comment]
By cache poisoning you mean that old versions of static files are not removed from Varnish?

**Comment by gberaudo** [comment]
No, I mean that an old version of the file is used instead of the correct one.

**Comment by asaunier** [comment]
I have updated the PR to take @tsauerwein 's suggestions.

Please note that the tool does not work for the ``node_modules`` and ``closure`` static views. I see that in GMF some special code is used: https://github.com/camptocamp/c2cgeoportal/blob/master/c2cgeoportal/__init__.py#L301-L318
Do we really need this? For instance if finally those ``node_modules`` files are also built together or downloaded from a CDN?

> How does it work in development mode with make serve? The cache is not used in that case?

Mmm you probably right, I guess the cache is used until you make a new commit. Perhaps that was the goal of the ``version`` parameter set either to 0 or nothing in ``config/dev`` or ``development.ini.in``... (not sure it would work out of the box though).
Anyway is it really a problem? When developing, don't you simply disable your browser cache?

**Comment by tsauerwein** [comment]
> No, I mean that an old version of the file is used instead of the correct one.

I thought Varnish by default takes query parameters into account?

**Comment by gberaudo** [comment]
Rebased and added support for debug mode and preventing cache poisoning.

**Comment by asaunier** [comment]
Looks good. Don't forget my question about the long URL here: https://github.com/c2corg/v6_ui/blob/cache_buster2/c2corg_ui/lib/cacheversion.py#L2-L3

**Pull request merged**